cleanlab_studio/internal/schema.py

```diff
@@ -0,0 +1,241 @@
+from dataclasses import dataclass
```

Comment:
I don't think it makes sense to have all of these internal classes and enums here. Is there any reason why we can't keep things super simple and just pass a dict of strs?

1. I don't like that we're maintaining this both in the CLI and the backend
2. This is additional complexity that we don't want the user to touch. We should just document a format ("pass in a dictionary of field names mapped to a tuple of data type, feature type") along w/ the supported data and feature types.

Reply:
further, we don't want to be doing _any_ validation here -- that's purely the responsibility of the studio servers

Reply:
Can't immediately think of any reasons against just using a dictionary instead of the Schema object here. Definitely agree that it's bad to have all these definitions duplicated in CLI and data ingestion. There are some backend dependencies on the version defined in CLI right now though, so if we remove this, we'll have to update those as well

Reply:
Most of the usage should go away with the data ingestion refactor. We're currently using the Schema object defined in CLI to [create simple zip schema](https://github.com/cleanlab/cleanlab-studio-backend/blob/982c28576668041334bfdaacf3745ebe53ad8a53/backend/api/upload/upload_api_helpers.py#L20) after the data ingestion refactor, but we could move the simple zip schema creation inside the data ingestion service instead. 

Other than that there's some references to CLI types in [training_service](https://github.com/cleanlab/cleanlab-studio-backend/blob/b56791f0dbef51ca1cbe144a96102bd9245ae0af/backend/services/training_service/training_service.py#L7) and [media_upload_service](https://github.com/cleanlab/cleanlab-studio-backend/blob/c2fc1c295419f4babde4f51d6a58c1b32649e5cc/backend/services/media_upload_service/media_upload_service.py)

Reply:
Got it, ok. Let's take this time to (finally) create a Cleanlab-studio-lib repo to hold this stuff. I'll do so this afternoon, and then we can remove this code from the CLI